src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -514,20 +528,23 @@ def make_positive(val: int) -> Column:
     if step > 1:
         # start can be negative --> make the lower-bound 0.
         step_bound_filter = (
-            (row_pos_col - greatest(pandas_lit(0), least(start, count_col - 1)))
+            (row_pos_col - greatest(pandas_lit(0), least(start, get_count_col() - 1)))
             % pandas_lit(step)
         ) == 0
     elif step < -1:
         # Similarly, stop can be too large --> make the upper-bound the max row number.
         step_bound_filter = (
-            (greatest(pandas_lit(0), least(stop, count_col - 1)) - row_pos_col)
+            (greatest(pandas_lit(0), least(stop, get_count_col() - 1)) - row_pos_col)
             % pandas_lit(abs(step))
         ) == 0
     else:  # abs(step) == 1, so all values in range are included.
         step_bound_filter = pandas_lit(True)
 
     filter_cond = left_bound_filter & right_bound_filter & step_bound_filter
     ordered_dataframe = frame.ordered_dataframe.filter(filter_cond)
+    if limit_n is not None:
+        # adding limit to exit scanning the whole table if enough result has been found
```

Comment:
maybe add some more details here like 
"""
add limit here could improve performance when the filter is applied directly on a table, because a limit can exit the scanning of table once enough record is found
"""